File: src/_01_CreationalPattern/_01_03_AbstractFactory/Whiteship.js
import Ship from "./Ship.js";

export default class Whiteship extends Ship {
  /**
   * @param { string } name
   */
  constructor(name) {
    super();

    this.setName(name);
    this.setType("whiteship");
    this.setColor("#fff");
    this.setLogo("🚀");
  }
}

File: src/_01_CreationalPattern/_01_03_AbstractFactory/java/before/Whiteship.java
package _01_CreationalPattern._01_03_AbstractFactory.java.before;

public class Whiteship extends Ship {
  
  public Whiteship(String name) {
    this.setName(name);
    this.setType("whiteship");
    this.setColor("#fff");
    this.setLogo("\uD83D\uDEE5️");
  }

}
